### PR TITLE
[SPARK-34587][CORE][SQL][DSTREAM][SS] Replaces map and getOrElse(false/true) with exists/forall

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Replaces `map and getOrElse(false/true)` with `exists/forall`, it's semantically consistent, but looks simpler.

**Before**

```
option.map(booleanFunc).getOrElse(false)
option.map(booleanFunc).getOrElse(true)
```

**After**

```
option.exists(booleanFunc)
option.forall(booleanFunc)
```

### Why are the changes needed?
Code Simpilefications.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

**Comment by LuciferYang** [comment]
To reduce conflicts, I also created PR for other branches:

- 3.1: https://github.com/apache/spark/pull/31701
- 3.0: https://github.com/apache/spark/pull/31702
- 2.4: https://github.com/apache/spark/pull/31703

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala [hunk 1]
```diff
@@ -357,10 +357,8 @@ class MicroBatchExecution(
   private def isNewDataAvailable: Boolean = {
     availableOffsets.exists {
       case (source, available) =>
-        committedOffsets
-          .get(source)
-          .map(committed => committed != available)
-          .getOrElse(true)
```
Comment (LuciferYang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala:363]:
```
committedOffsets
          .get(source)
          .forall(committed => committed != available)
```

can be further simplified as

```
!committedOffsets
          .get(source).contains(available)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala [hunk 2]
```diff
@@ -53,31 +53,31 @@ private[sql] class JSONOptions(
   val samplingRatio =
     parameters.get("samplingRatio").map(_.toDouble).getOrElse(1.0)
   val primitivesAsString =
-    parameters.get("primitivesAsString").map(_.toBoolean).getOrElse(false)
+    parameters.get("primitivesAsString").exists(_.toBoolean)
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:56]: I know this calls less but I wonder if it is really more readable. To me `map(_).getOrElse(false)` looks more readable: if exists convert to a bool, and defaults `false`.

Comment (LuciferYang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala:56]: Got it ~ I will close this pr ~ thx ~

**Pull request closed without merging**